src/openzeppelin/token/erc1155/library.cairo

```diff
@@ -0,0 +1,600 @@
+    ):
+        alloc_locals
+        let (caller) = get_caller_address()
+        with_attr error_message("ERC1155: called from zero address"):
+            assert_not_zero(caller)
+        end
+        with_attr error_message("ERC1155: burn from the zero address"):
+            assert_not_zero(from_)
+        end
+
+        # Deduct from burner
+        let (from_balance : Uint256) = ERC1155_balances.read(id=id, account=from_)
+        with_attr error_message("ERC1155: burn amount exceeds balance"):
+            let (new_balance : Uint256) = uint256_checked_sub_le(from_balance, amount)
+        end
+        ERC1155_balances.write(id=id, account=from_, value=new_balance)
+
+        let (operator) = get_caller_address()
+        TransferSingle.emit(operator=operator, from_=from_, to=0, id=id, value=amount)
+        return ()
+    end
+
+    func _burn_batch{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        from_ : felt, ids_len : felt, ids : Uint256*, amounts_len : felt, amounts : Uint256*
+    ):
+        alloc_locals
+        let (caller) = get_caller_address()
+        with_attr error_message("ERC1155: called from zero address"):
+            assert_not_zero(caller)
+        end
+        with_attr error_message("ERC1155: burn from the zero address"):
+            assert_not_zero(from_)
+        end
+        # Check args are equal length arrays
+        with_attr error_message("ERC1155: ids and amounts length mismatch"):
+            assert ids_len = amounts_len
+        end
+        # Recursive call
+        let len = ids_len
+        burn_batch_iter(from_, len, ids, amounts)
+        let (operator) = get_caller_address()
+        TransferBatch.emit(
+            operator=operator,
+            from_=from_,
+            to=0,
+            ids_len=ids_len,
+            ids=ids,
+            values_len=amounts_len,
+            values=amounts,
+        )
+        return ()
+    end
+
+    #
+    # Internals
+    #
+
+    func _set_approval_for_all{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        owner : felt, operator : felt, approved : felt
+    ):
+        # check approved is bool
+        assert approved * (approved - 1) = 0
+        # caller/owner already checked  non-0
+        with_attr error_message("ERC1155: setting approval status for zero address"):
+            assert_not_zero(operator)
+        end
+        with_attr error_message("ERC1155: setting approval status for self"):
+            assert_not_equal(owner, operator)
+        end
+        ERC1155_operator_approvals.write(owner, operator, approved)
+        ApprovalForAll.emit(owner, operator, approved)
+        return ()
+    end
+
+    func _setURI{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(newuri : felt):
+        ERC1155_uri.write(newuri)
+        return ()
+    end
+
+    func _do_safe_transfer_acceptance_check{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    }(
```

Comment:
Following the forthcoming extensibility pattern update, I believe we should move this outside of the namespace since it should not be accessed from outside the library.